backend/apps/progress/views.py

```
@@ -811,6 +811,31 @@ class CertificateVerificationThrottle(AnonRateThrottle):
     rate = "10/minute"
 
 
+class CertificateVerifyView(APIView):
+    """Public API to verify certificate by hash."""
+    
+    permission_classes = []  # Public endpoint
+    
+    def get(self, request, hash):
```

Ruff (0.15.21)

[error] Function argument `hash` is shadowing a Python builtin

(A002)